src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs

```diff
@@ -0,0 +1,98 @@
+{
+    private static readonly LocalizableResourceString Title = new(nameof(Resources.TestClassShouldHaveTestMethodTitle), Resources.ResourceManager, typeof(Resources));
+    private static readonly LocalizableResourceString Description = new(nameof(Resources.TestClassShouldHaveTestMethodDescription), Resources.ResourceManager, typeof(Resources));
+    private static readonly LocalizableResourceString MessageFormat = new(nameof(Resources.TestClassShouldHaveTestMethodMessageFormat), Resources.ResourceManager, typeof(Resources));
+
+    internal static readonly DiagnosticDescriptor TestClassShouldHaveTestMethodRule = DiagnosticDescriptorHelper.Create(
+        DiagnosticIds.TestClassShouldHaveTestMethodRuleId,
+        Title,
+        MessageFormat,
+        Description,
+        Category.Design,
+        DiagnosticSeverity.Info,
+        isEnabledByDefault: true);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
+       = ImmutableArray.Create(TestClassShouldHaveTestMethodRule);
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+        context.EnableConcurrentExecution();
+
+        context.RegisterCompilationStartAction(context =>
+        {
+            if (context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingTestClassAttribute, out var testClassAttributeSymbol))
+            {
+                var testMethodAttributeSymbol = context.Compilation.GetTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingTestMethodAttribute);
+                var assemblyInitializationAttributeSymbol = context.Compilation.GetTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingAssemblyInitializeAttribute);
+                var assemblyCleanupAttributeSymbol = context.Compilation.GetTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingAssemblyCleanupAttribute);
+                context.RegisterSymbolAction(
+                    context => AnalyzeSymbol(context, testClassAttributeSymbol, testMethodAttributeSymbol, assemblyInitializationAttributeSymbol, assemblyCleanupAttributeSymbol),
+                    SymbolKind.NamedType);
+            }
+        });
+    }
+
+    private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol testClassAttributeSymbol, INamedTypeSymbol? testMethodAttributeSymbol,
+        INamedTypeSymbol? assemblyInitializationAttributeSymbol, INamedTypeSymbol? assemblyCleanupAttributeSymbol)
+    {
+        var namedTypeSymbol = (INamedTypeSymbol)context.Symbol;
+
+        bool isTestClass = false;
+        foreach (var classAttribute in namedTypeSymbol.GetAttributes())
+        {
+            if (classAttribute.AttributeClass.Inherits(testClassAttributeSymbol))
+            {
+                isTestClass = true;
```

Comment:
can we exit if true?

Reply:
I need to check the test method as well

Reply:
oh yup we can I thought it's the other forech